src/core/src/TKFactoryRandomWalk.C

```diff
@@ -0,0 +1,48 @@
+//-----------------------------------------------------------------------bl-
+//--------------------------------------------------------------------------
+//
+// QUESO - a library to support the Quantification of Uncertainty
+// for Estimation, Simulation and Optimization
+//
+// Copyright (C) 2008-2015 The PECOS Development Team
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the Version 2.1 GNU Lesser General
+// Public License as published by the Free Software Foundation.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc. 51 Franklin Street, Fifth Floor,
+// Boston, MA  02110-1301  USA
+//
+//-----------------------------------------------------------------------el-
+
+#include <queso/TKFactoryRandomWalk.h>
+#include <queso/ScaledCovMatrixTKGroup.h>
+
+namespace QUESO
+{
+
+template <class DerivedTK>
+SharedPtr<BaseTKGroup<GslVector, GslMatrix> >::Type
+TKFactoryRandomWalk<DerivedTK>::build_tk()
+{
+  SharedPtr<BaseTKGroup<GslVector, GslMatrix> >::Type new_tk;
+
+  new_tk.reset(new DerivedTK(this->m_options->m_prefix.c_str(),
+                             *(this->m_vectorSpace),
+                             *(this->m_dr_scales),
+                             *(this->m_initial_cov_matrix)));
+
+  return new_tk;
```

Comment:
This could be a one-liner.  The compiler ought to be able to optimize out the redundancies, though, so if you think this is more readable it's fine.